### DISCUSSION TITLE
Data containing empty or nan

### DISCUSSION THREAD

**Discussion by anthonysirico** [pyg-team/pytorch_geometric#8076] [Q&A] [answered]
Here is what the data looks like:

![image](https://github.com/pyg-team/pytorch_geometric/assets/100864754/6b6d4627-b212-48ac-a762-745d0b329e21)

The first column "Adjacency Matrix" is to create the graphs (obviously), the other 5 columns are graph-level features. 
1. The NaN and empty cells is significant to the data, it tell us that the graph did not simulate. Can PyG/PyTorch read empty or NaN? 
2. If yes to both, which is better NaN or empty?
3. If not, any suggestions to replace the value?

**Comment by anthonysirico** [comment, marked as answer]
https://github.com/pyg-team/pytorch_geometric/discussions/7461

**Comment by rusty1s** [comment]
Yes, PyTorch can handle NaNs in tensor data. However, gradients will be NaN in this case, so you need a way to impute it before using it as input into a neural network.

**Reply by anthonysirico** [reply]
thanks, I ended up using `torch.nan_to_num`
